### PR TITLE
[API] Add glorotUniform init and clean up other random init code

### PR DESCRIPTION
**Pull request by jekbradbury**
Removes some uses of `_TFHoistable`; will remove the rest of it later.

### CODE REVIEW

**Review by rxwei — APPROVED** [review]

File: stdlib/public/TensorFlow/Tensor.swift [hunk 1]
```diff
@@ -673,16 +659,12 @@ public extension Tensor where Scalar : BinaryFloatingPoint,
...
-        }
-        return _TFTensorFromScalars(scalars, shape: shape.dimensions)
-      }
-    ).toAccelerator()
+    let dist = UniformFloatingPointDistribution<Scalar>()
+    var scalars: [Scalar] = []
+    for _ in 0 ..< shape.contiguousSize {
+      scalars.append(dist.next(using: &generator))
+    }
+    self = Tensor(scalars, shape: shape.dimensions)
```
Comment (rxwei) [inline stdlib/public/TensorFlow/Tensor.swift]: same here

File: stdlib/public/TensorFlow/Tensor.swift [hunk 2]
```diff
@@ -722,17 +693,12 @@ public extension Tensor where Scalar : BinaryFloatingPoint,
...
-        }
-        return _TFTensorFromScalars(scalars, shape: shape.dimensions)
-      }
-    ).toAccelerator()
+    let dist = NormalDistribution<Scalar>(mean: mean, standardDeviation: stddev)
+    var scalars: [Scalar] = []
+    for _ in 0 ..< shape.contiguousSize {
+     scalars.append(dist.next(using: &generator))
+    }
+    self = Tensor(scalars, shape: shape.dimensions)
```
Comment (rxwei) [inline stdlib/public/TensorFlow/Tensor.swift]: same here

**Comment by jekbradbury** [comment]
Addressed comments. Can you start CI?

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow macOS

**Comment by rxwei** [comment]
@swift-ci please test tensorflow macOS

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by jekbradbury** [comment]
Hmm, `no_copy.swift` is explicitly testing the GPE behavior that we're killing in this PR, but the failure also demonstrates that `init(randomUniform:)` and friends simply don't work with GPE if that behavior is killed. I don't know what the current policy on GPE is; creating a random tensor seems like a fairly fundamental operation that should continue to work in GPE mode as long as we care about it at all?

**Review by dan-zheng — APPROVED** [review]
Nice!

**Comment by rxwei** [comment]
We’ve mentioned that in the last team meeting and the conclusion was if any GPE tests are blocking we should remove then.

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**